### PR TITLE
add fireCheck before shooting and fireRate to weapons

### PR DESCRIPTION
**Pull request by MTattrie**
Add fireState to Weapon (ready to fire or not ready to fire)
Add fireRate to Weapons
Shooting first checks it weapon is ready to fire before shootin

Signed-off-by: Mark Tattrie <MTattrie93@gmail.com>

### CODE REVIEW

**Comment by MTattrie** [comment]
Add fireState to Weapon (ready to fire or not ready to fire)
Add fireRate to Weapons
Shooting first checks it weapon is ready to fire before shooting

File: src/Entity.cpp [hunk 1]
```diff
@@ -46,10 +46,10 @@ void Entity::setPosition(float px, float py) {
...
 
 void Entity::updateHitBoxes() {
-	movementHitBox->move(x, y);
-	projectileHitBox->move(x, y);
-	damageHitBox->move(x, y);
-	pickupHitBox->move(x, y);
+ movementHitBox->move(x, y);
+ projectileHitBox->move(x, y);
+ damageHitBox->move(x, y);
+ pickupHitBox->move(x, y);
```
Comment (isaacmorneau) [inline src/Entity.cpp]: a single space, while it isnt tabs, is not a valid indentation level. use 4 spaces

File: src/weapons/HandGun.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 */
 #include "HandGun.h"
 
-HandGun::HandGun() : InstantWeapon(HandgunVars::TYPE, HandgunVars::RANGE, HandgunVars::DAMAGE, 
-        HandgunVars::CLIP, HandgunVars::CLIPMAX, HandgunVars::AMMO, HandgunVars::AOE, HandgunVars::RELOAD) {
+HandGun::HandGun() : InstantWeapon(HandgunVars::TYPE, HandgunVars::RANGE, HandgunVars::DAMAGE,
+        HandgunVars::CLIP, HandgunVars::CLIPMAX, HandgunVars::AMMO, HandgunVars::AOE, HandgunVars::RELOAD, HandgunVars::FIRERATE, HandgunVars::READY) {
```
Comment (isaacmorneau) [inline src/weapons/HandGun.cpp]: line is too long, should be less than 100

File: src/weapons/HandGun.h [hunk 3]
```diff
@@ -15,9 +15,11 @@ namespace HandgunVars {
     const int DAMAGE   = 500;
     const int CLIP     = 2;
     const int CLIPMAX  = 2;
-    const int AMMO     = 1000;
+    const int AMMO     = 50;
     const int AOE      = 1;
     const int RELOAD   = 3;
+    const int FIRERATE = 1000;
+    const int READY    = true;
```
Comment (isaacmorneau) [inline src/weapons/HandGun.h]: use `constexpr` instead of `const` for defaults

File: src/weapons/InstantWeapon.cpp [hunk 4]
```diff
@@ -10,24 +10,24 @@
 #include <stdio.h>
 #include <iostream>
 
-InstantWeapon::InstantWeapon(std::string type, int range, int damage, 
-        int clip, int clipMax, int ammo, int AOE, int reloadSpeed)
-: Weapon(type, range, damage, clip, clipMax, ammo, AOE, reloadSpeed) {
+InstantWeapon::InstantWeapon(std::string type, int range, int damage,
+        int clip, int clipMax, int ammo, int AOE, int reloadSpeed, int fireRate, bool isReadyToFire)
+: Weapon(type, range, damage, clip, clipMax, ammo, AOE, reloadSpeed, fireRate, isReadyToFire) {
```
Comment (JohnAgapeyev) [inline src/weapons/InstantWeapon.cpp]: Line should be indented since it is a wrapped line

File: src/weapons/HandGun.h [hunk 5]
```diff
@@ -15,9 +15,11 @@ namespace HandgunVars {
     const int DAMAGE   = 500;
     const int CLIP     = 2;
     const int CLIPMAX  = 2;
-    const int AMMO     = 1000;
+    const int AMMO     = 50;
     const int AOE      = 1;
     const int RELOAD   = 3;
+    const int FIRERATE = 1000;
```
Comment (JohnAgapeyev) [inline src/weapons/HandGun.h]: Change const to constexpr

File: src/weapons/InstantWeapon.cpp [hunk 6]
```diff
@@ -45,5 +45,25 @@ void InstantWeapon::fire(Marine &marine){
...
     //=======================================================
-    
+}
+
+void InstantWeapon::melee(Marine &marine){
+    printf("InstantWeapon::melee\n");
+
+    CollisionHandler ch = GameManager::instance()->getCollisionHandler();
+
+    int x = marine.getX();
```
Comment (JohnAgapeyev) [inline src/weapons/InstantWeapon.cpp]: Use const if not modified

File: src/weapons/Rifle.h [hunk 7]
```diff
@@ -17,6 +17,8 @@ namespace RifleVars {
     const int AMMO     = 1000;
     const int AOE      = 1;
     const int RELOAD   = 3;
+    const int FIRERATE = 1000;
+    const int READY    = true;
```
Comment (JohnAgapeyev) [inline src/weapons/Rifle.h]: Use constexpr

File: src/weapons/ShotGun.h [hunk 8]
```diff
@@ -18,6 +18,8 @@ namespace ShotgunVars {
     const int AMMO     = 1000;
     const int AOE      = 1;
     const int RELOAD   = 3;
+    const int FIRERATE = 600;
+    const int READY    = true;
```
Comment (JohnAgapeyev) [inline src/weapons/ShotGun.h]: constexpr

**Pull request merged**